pkg/permissions/permissions.go

```diff
@@ -435,25 +434,24 @@ func GetPermissionsByType(db couchdb.Database, doctype string, cursor couchdb.Cu
 }
 
 // GetSharedWithMePermissionsByDoctype retrieves the permissions in all
-// the sharings that apply to the given doctype, where the user is a recipient.
+// the sharings that apply to the given doctype, where the user is a recipient
+// (i.e. owner is false).
 //
 // The cursor will be modified in place.
 func GetSharedWithMePermissionsByDoctype(db couchdb.Database, doctype string, cursor couchdb.Cursor) ([]*Permission, error) {
-	return getSharedWithPermissionsByDoctype(db, doctype, cursor,
-		consts.SharedWithMePermissionsView)
+	return getSharedWithPermissionsByDoctype(db, doctype, cursor, false)
 }
 
 // GetSharedWithOthersPermissionsByDoctype retrieves the permissions in all the
-// sharings that apply to the given doctype, where the user is the sharer.
+// sharings that apply to the given doctype, where the user is the sharer (i.e.
+// owner is true).
 //
 // The cursor will be modified in place.
 func GetSharedWithOthersPermissionsByDoctype(db couchdb.Database, doctype string, cursor couchdb.Cursor) ([]*Permission, error) {
-	return getSharedWithPermissionsByDoctype(db, doctype, cursor,
-		consts.SharedWithOthersPermissionsView)
+	return getSharedWithPermissionsByDoctype(db, doctype, cursor, true)
 }
 
-func getSharedWithPermissionsByDoctype(db couchdb.Database, doctype string, cursor couchdb.Cursor, view *couchdb.View) ([]*Permission, error) {
-
+func getSharedWithPermissionsByDoctype(db couchdb.Database, doctype string, cursor couchdb.Cursor, owner bool) ([]*Permission, error) {
 	var req = &couchdb.ViewRequest{
 		StartKey:    []string{doctype},
 		EndKey:      []string{doctype, couchdb.MaxString},
```

Comment:
If you invert owner & sharing_id in view definition, you could filter directly here instead of in the for loop below.